tests/app/views/test_agreements.py

```diff
@@ -0,0 +1,264 @@
+            'id': agreement_id,
+            'supplierId': supplier_framework['supplier_id'],
+            'frameworkId': supplier_framework['framework_id'],
+            'signedAgreementDetails': {'details': 'here'},
+            'signedAgreementPath': 'path',
+            'signedAgreementReturnedAt': '2016-10-01T01:01:01.000000Z',
+            'countersignedAgreementDetails': {'countersigneddetails': 'here'},
+            'countersignedAgreementReturnedAt': '2016-11-01T01:01:01.000000Z',
+            'countersignedAgreementPath': '/example.pdf'
+        }
+
+
+class TestUpdateFrameworkAgreement(BaseApplicationTest):
+    def create_agreement(self, supplier_framework, **framework_agreement_kwargs):
+        with self.app.app_context():
+            agreement = FrameworkAgreement(
+                supplier_id=supplier_framework['supplier_id'],
+                framework_id=supplier_framework['framework_id'],
+                **framework_agreement_kwargs)
+            db.session.add(agreement)
+            db.session.commit()
+
+            return agreement.id
+
+    def post_agreement_update(self, agreement_id, agreement):
+        return self.client.post(
+            '/agreements/{}'.format(agreement_id),
+            data=json.dumps(
+                {
+                    'updated_by': 'interested@example.com',
+                    'agreement': agreement
+                }),
+            content_type='application/json')
+
+    def test_cant_set_agreement_details_for_framework_without_agreement_version(self, supplier_framework):
+        agreement_id = self.create_agreement(supplier_framework)
+        res = self.client.post(
+            '/agreements/{}'.format(agreement_id),
+            data=json.dumps(
+                {
+                    'updated_by': 'interested@example.com',
+                    'agreement': {
+                        'signerAgreementDetails': {
+                            'signerName': 'name',
+                            'signerRole': 'role',
+                        }
+                    }
+                }),
+            content_type='application/json')
+
+        assert res.status_code == 400
+
+    def test_can_update_framework_agreement_details(self, supplier_framework):
+        agreement_id = self.create_agreement(supplier_framework)
+        res = self.post_agreement_update(agreement_id, {
+            'signedAgreementDetails': {
+                'signerName': 'name',
+                'signerRole': 'role',
+            }
+        })
+
+        assert res.status_code == 200
+        data = json.loads(res.get_data(as_text=True))
+        assert data['agreement'] == {
+            'id': agreement_id,
+            'supplierId': supplier_framework['supplier_id'],
+            'frameworkId': supplier_framework['framework_id'],
+            'signedAgreementDetails': {
+                'signerName': 'name',
+                'signerRole': 'role',
+            }
+        }
```

Comment:
Is it worth also doing a subsequent GET here to check that the update has actually saved, rather than just testing the initial response to the POST? (Thanks to @risicle I have a feeling this is something we're generally guilty of not testing enough on our update routes.)